Makefile

```diff
@@ -29,10 +29,52 @@ RESTY_LUAROCKS_VERSION ?= `grep RESTY_LUAROCKS_VERSION $(KONG_SOURCE_LOCATION)/.
 RESTY_OPENSSL_VERSION ?= `grep RESTY_OPENSSL_VERSION $(KONG_SOURCE_LOCATION)/.requirements | awk -F"=" '{print $$2}'`
 RESTY_PCRE_VERSION ?= `grep RESTY_PCRE_VERSION $(KONG_SOURCE_LOCATION)/.requirements | awk -F"=" '{print $$2}'`
 KONG_BUILD_TOOLS ?= '4.3.2'
-KONG_VERSION ?= `cat $(KONG_SOURCE_LOCATION)/kong-*.rockspec | grep tag | awk '{print $$3}' | sed 's/"//g'`
 OPENRESTY_PATCHES_BRANCH ?= master
 KONG_NGINX_MODULE_BRANCH ?= master
 
+PACKAGE_TYPE ?= deb
+REPOSITORY_NAME ?= kong-${PACKAGE_TYPE}
+REPOSITORY_OS_NAME ?= ${RESTY_IMAGE_BASE}
+KONG_PACKAGE_NAME ?= kong
+KONG_VERSION ?= `cat $(KONG_SOURCE_LOCATION)/kong-*.rockspec | grep -m1 tag | awk '{print $$3}' | sed 's/"//g'`
+
+BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+ISTAG = false
+ISPRERELEASE = false
+ifeq ($(BRANCH),HEAD)
+	ISTAG = true
+	BRANCH = $(shell git describe --tags --abbrev=0)
+	POSSIBLE_VERSION = $(shell git describe --tags --abbrev=0 | awk -F"-" '{print $$1}')
+	POSSIBLE_PRERELEASE_NAME = $(shell git describe --tags --abbrev=0 | awk -F"-" '{print $$2}')
+else
+	REPOSITORY_NAME = kong-${BRANCH}
+	REPOSITORY_OS_NAME = ${BRANCH}
+	KONG_PACKAGE_NAME = kong-${BRANCH}
+	KONG_VERSION = `date +%Y-%m-%d`
+endif
+ifneq ($(POSSIBLE_PRERELEASE_NAME),)
+	KONG_VERSION = ${BRANCH}
+	REPOSITORY_NAME = kong-prerelease
+endif
+
+release:
+ifeq ($(ISTAG),false)
+	sed -i -e '/return string\.format/,/\"\")/c\return "$(KONG_VERSION)\"' kong/meta.lua
```

Comment:
if we're releasing a branch (presumably this is a nightly build) we need to muck with the `meta.lua`

Reply:
This seems very brittle. It means we can break the release process if we change the way we format that particular string inside meta.lua. In particular, we often use a shorthand local variable called `fmt` instead of `string.format`.

I'm not sure about what's the best solution here. One could be to just add a comment to meta: "hey don't change the following line or you might break the release". Another possibility could be to replace the whole meta.lua file by a custom one.

Reply:
would it make it better or worse if I point out we've been doing trickery like this for a while 😆 

Propose: we add the comment and try to circle back at a later date for something more elegant / less fragile?

Reply:
Yeah a comment in meta.lua is fast and will solve the issue for now.

Reply:
Added the comment in 67e9b4f